components/menu/menu.tsx

```diff
@@ -0,0 +1,259 @@
+        event.preventDefault();
+
+        if (isMobileView) {
+            dispatch(
+                openModal<MenuModalProps>({
+                    modalId: `${props.menuButton.id}-${props.menu.id}`,
+                    dialogType: MenuModal,
+                    dialogProps: {
+                        menuButtonId: props.menuButton.id,
+                        menuId: props.menu.id,
+                        menuAriaLabel: props.menu?.['aria-label'] ?? '',
+                        children: props.children,
+                    },
+                }),
+            );
+        } else {
+            setAnchorElement(event.currentTarget);
+        }
+    }
+
+    function handleMenuButtonMouseDown() {
+        // This is needed to prevent focus-visible being set on clicking menuitems with mouse
+        setDisableAutoFocusItem(true);
+    }
+
+    function renderMenuButton() {
+        // We construct the menu button so we can set onClick correctly here to support both web and mobile view
+        const triggerElement = (
+            <button
+                id={props.menuButton.id}
+                aria-controls={props.menu.id}
+                aria-haspopup={true}
+                aria-expanded={isMenuOpen}
+                aria-label={props.menuButton?.['aria-label'] ?? ''}
+                className={props.menuButton?.class ?? ''}
+                onClick={handleMenuButtonClick}
+                onMouseDown={handleMenuButtonMouseDown}
+            >
+                {props.menuButton.children}
+            </button>
+        );
+
+        if (props.menuButtonTooltip && props.menuButtonTooltip.text && !isMobileView) {
+            return (
+                <OverlayTrigger
+                    delayShow={OVERLAY_TIME_DELAY}
+                    placement={props?.menuButtonTooltip?.placement ?? 'top'}
+                    overlay={
+                        <Tooltip
+                            id={props.menuButtonTooltip.id}
+                            className={props.menuButtonTooltip?.class ?? ''}
+                        >
+                            {props.menuButtonTooltip.text}
+                        </Tooltip>
+                    }
+                    disabled={isMenuOpen}
+                >
+                    {triggerElement}
+                </OverlayTrigger>
+            );
+        }
+
+        return triggerElement;
+    }
+
+    useEffect(() => {
+        if (props.menu.onToggle) {
+            props.menu.onToggle(isMenuOpen);
+        }
+    }, [isMenuOpen]);
+
+    if (isMobileView) {
+        // In mobile view, the menu is rendered as a modal
+        return renderMenuButton();
+    }
+
+    return (
+        <CompassDesignProvider theme={theme}>
+            {renderMenuButton()}
+            <MuiMenuStyled
+                anchorEl={anchorElement}
+                open={isMenuOpen}
+                onClose={handleMenuClose}
+                onClick={handleMenuClick}
+                onKeyDown={handleMenuKeyDown}
```

Comment:
Can we make sure onKeyDown can be called from outside as well? In the dot menu component, I am thinking of using it for keyboard shortcuts instead of adding an event listener
<img width="568" alt="Screenshot 2023-01-17 at 12 13 50 AM" src="https://user-images.githubusercontent.com/16203333/212747282-47e2a05c-d6e6-4943-8227-9bc6f043d7d4.png">

Reply:
When we add an event listener, events are from dom. But this they remain React synthetic events

Reply:
This is interesting, let me verify that before we merge the pr, but we can always improve on this component as we integrate it in more places.